test/test/multi_threaded_common.h

```diff
@@ -237,6 +237,17 @@ namespace concurrent_collections
     {
         return it += offset;
     }
+
+    // Dummy definition to keep compiler happy. Instantiated by the
+    // constructor whose purpose is to offer an implicit conversion from
+    // non-const iterator to const iterator. When applied to a const
+    // iterator, it offers implicit construction from this garbage type
+    // to const iterator (triggering its instantiation), but then the constructor
+    // is deleted by SFINAE, so the type is never actually created at runtime.
+    template<typename Container>
```

Comment:
I sort of forgot to mention why we need this: If we didn't specialize, then the regular version above would be instantiated, but it doesn't work when `Iterator` is void!

Reply:
Let's update the comment - that helps. 😉